quickfixj-core/src/main/java/quickfix/mina/SessionConnector.java

```diff
@@ -318,7 +318,7 @@ protected void startSessionTimer() {
         if (shortLivedExecutor != null) {
             timerTask = new DelegatingTask(timerTask, shortLivedExecutor);
         }
-        sessionTimerFuture = scheduledExecutorService.scheduleAtFixedRate(timerTask, 0, 1000L,
+        sessionTimerFuture = SCHEDULED_EXECUTOR.scheduleAtFixedRate(timerTask, 0, 1000L,
```

Comment:
Not directly related to this change, but we probably don't want to queue up timer tasks if they take too long for some reason.

https://github.com/quickfix-j/quickfixj/blob/e53d41bc6b7a51ae02a63b7b0222846d248dcb7c/quickfixj-core/src/main/java/quickfix/mina/initiator/IoSessionInitiator.java#L364

Reply:
The `reconnectTasks` timed out after a maximum of 2000ms. That was the reason to have a bigger pool to service these requests, see #254 .
What do you suggest to prevent piling up these tasks?

Reply:
Similar to `IoSessionInitiator`. As a rule of thumb `scheduleWithFixedDelay` is better in majority of cases than `scheduleAtFixedRate`. It respects the delay, but it only reschedules when the tasks finishes - this does not allow bursts if for some reason task is slow.